src/uu/ls/src/ls.rs

```diff
@@ -858,6 +881,11 @@ fn sort_entries(entries: &mut Vec<PathBuf>, config: &Config) {
         // The default sort in GNU ls is case insensitive
         Sort::Name => entries.sort_by_key(|k| k.to_string_lossy().to_lowercase()),
         Sort::Version => entries.sort_by(version_cmp::version_cmp),
+        Sort::Extension => entries.sort_by(|a, b| {
+            a.extension()
+                .cmp(&b.extension())
+                .then(a.to_string_lossy().cmp(&b.to_string_lossy()))
```

Comment:
I believe Rust always evaluates the arguments to functions, even if they don't end up being used. So
```
.then_with(|| a.to_string_lossy().cmp(&b.to_string_lossy())
```
_could_ be more performant. I'm not 100%  sure though.